src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java

```diff
@@ -575,8 +578,8 @@ public Runnable lockStudentRepositories(ProgrammingExercise exercise, Predicate<
         return () -> {
             SecurityUtils.setAuthorizationObject();
             try {
-                List<ProgrammingExerciseStudentParticipation> failedLockOperations = removeWritePermissionsFromAllStudentRepositories(programmingExerciseId, condition);
-                stashStudentChangesAndNotifyInstructor(exercise, failedLockOperations.size(), condition);
+                var failedLockOperationsFuture = removeWritePermissionsFromAllStudentRepositories(programmingExerciseId, condition);
+                failedLockOperationsFuture.thenAccept(failedLockOperations -> stashStudentChangesAndNotifyInstructor(exercise, failedLockOperations.size(), condition));
```

Comment:
```suggestion
                var failedLockOperations = removeWritePermissionsFromAllStudentRepositories(programmingExerciseId, condition);
                failedLockOperations.thenAccept(failures -> stashStudentChangesAndNotifyInstructor(exercise, failures.size(), condition));
```
...to be consistent with lines 557-558 and 605-606. It would also be fine by me to change the lines there to look the same.